### PR TITLE
engine: move tiltfile execution outside of init action

### PR DESCRIPTION
**Pull request by jazzdan**

### CODE REVIEW

File: internal/engine/upper.go [hunk 1]
```diff
@@ -872,6 +892,6 @@ func handleDockerComposeLogAction(state *store.EngineState, action DockerCompose
 	ms.ResourceState = dcState.WithCurrentLog(append(dcState.CurrentLog, action.Log...))
 }
 
-func handleTiltfileLogAction(state *store.EngineState, action TiltfileLogAction) {
+func handleTiltfileLogAction(ctx context.Context, state *store.EngineState, action TiltfileLogAction) {
```
Comment (nicks) [inline internal/engine/upper.go]: why do you need a context here?

File: internal/tiltfile/files.go [hunk 2]
```diff
@@ -238,6 +238,7 @@ func (s *tiltfileState) local(thread *starlark.Thread, fn *starlark.Builtin, arg
 		return nil, err
 	}
 
+	s.logger.Printf("Running `%s`\n", command)
```
[the remark this answers is not in the document]
Comment (jazzdan) [reply internal/tiltfile/files.go]: This is just to give us more information in the Tiltfile log. This is what gives you insight in to what's going on when there's a `local` that runs for a long time and causes the Tiltfile to take a long time to finish executing.

File: internal/engine/upper.go [hunk 3]
```diff
@@ -452,8 +455,14 @@ func handleConfigsReloadStarted(
 	state *store.EngineState,
 	event ConfigsReloadStartedAction,
 ) {
-	state.CurrentTiltfileBuild = model.BuildRecord{}
 	state.PendingConfigFileChanges = make(map[string]bool)
+	status := model.BuildRecord{
+		StartTime: event.StartTime,
+		Reason:    model.BuildReasonFlagConfig,
+		Edits:     []string{state.TiltfilePath},
```
[the remark this answers is not in the document]
Comment (jazzdan) [reply internal/engine/upper.go]: True, but it's all config files, not just the ones that changed

https://github.com/windmilleng/tilt/blob/82a0e172d3a0e7bcf36163baa8d79aa3c5702bc6/internal/engine/configs_controller.go#L58

**Comment by jazzdan** [comment]
@nicks @landism this is ready for review now, sorry for the confusion!

File: internal/tiltfile/files.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (nicks) [reply internal/tiltfile/files.go]: :+1: 

i would probably use %q so that you don't have to worry about escaping messing up the readability

**Pull request merged**